shared/src/main/scala/io/lemonlabs/uri/Uri.scala

```diff
@@ -953,6 +974,20 @@ final case class AbsoluteUrl(scheme: String,
 
   private[uri] def toString(c: UriConfig, hostToString: Host => String): String =
     scheme + "://" + authority.toString(c, hostToString) + path.toString(c) + queryToString(c) + fragmentToString(c)
+
+  def normalize(removeEmptyPathParts: Boolean = false, slashTerminated: Option[Boolean] = None): Self = {
+    val scheme = this.scheme.toLowerCase
+    val isHttp = scheme == "http" || scheme == "https"
+    copy(
+      scheme = scheme,
+      authority = authority.normalize(if (isHttp) Some(scheme) else None),
```

Comment:
We can probably remove the `isHttp` check here and just always pass in `Some(scheme)`? Then the presence (or not) of the scheme in `Authority.DefaultPorts` will determine if a default port gets removed